src/frontend/mame/ui/menu.h

```diff
@@ -64,6 +64,7 @@ class menu
 
 	// append a new item to the end of the menu
 	void item_append(const char *text, const char *subtext, UINT32 flags, void *ref, menu_item_type type = menu_item_type::UNKNOWN);
+	void item_append(const std::string &text, const std::string &subtext, UINT32 flags, void *ref, menu_item_type type = menu_item_type::UNKNOWN);
```

Comment:
You're going to want to provide a method taking `std::string &&` as well to avoid unnecessary allocation/copying.  You may as well get rid of the `char *` version since there's no way the menu can use long-lived C strings without copying now anyway.